lib/diagnostic.js

```diff
@@ -8,38 +8,57 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-const min = function min() {
-
+const min = function min(array, init) {
+  let result;
+  if (init !== undefined){
+    result = init;
+  }
+  return result;
 };
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-const sum = function sum() {
 
+const sum = function(prev, curr){
+  return prev + curr;
 };
 
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
 const any = function any(array, callback) {
-
+  for (let i = 0; i < array.length; i++) {
+    if(array[i]){
+      return true;
+    }
+  }
 };
 
 // Question 4
 // Write a function that returns a new array
 // made up of the array elements for which invoking
 // the callback returns a truthy value
 const select = function select(array, callback) {
-
+  let array2 = [];
+  for (let i = 0; i < array.length; i++) {
+    if(callback(array[i])){
+      array2.push(array[i]);
+    }
+  }
 };
 
```

Comment:
Similar issue here - very close, but you'd need to remember to return the array you've accumulated values into.